charts/jenkins/ci/other-values.yaml

```diff
@@ -4,6 +4,7 @@ controller:
   runAsUser: 0
   fsGroup: 1000
   JCasC:
+    secret: true
```

Comment:
Could you explain the use case more? Jcasc has plenty of secret engines you shouldn’t need to pass anything in as plain text

Reply:
it's an easy way to protect credentials and other secrets for users who don't have the nuances of JCasC figured out

Reply:
better to just do it properly I think:
https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/features/secrets.adoc

There's already `controller.additionalSecrets` to handle this usecase properly

Reply:
That makes sense. I'll talk to my team about whether we can use this. I don't love the idea of giving secret access to apps whenever possible, but i may be able to restrict it to specifically the credentials secrets.

I guess the upside to this PR is that it makes quick-and-dirty setups a bit safer, while the downside is that it may encourage people to not pursue other options. Perhaps the middle ground could be to add a comment to `values.yaml` explaining the kubernetes credentials plugin

Reply:
at a minimum I'd suggest adding something to the docs explaining this, so that people aren't tempted to put credentials in the configmap (and not even necessarily even knowing it's going into a configmap). Jenkins can be quite intimidating and this is an easy mistake to make

Reply:
So, if i understand it correctly, [kubernetes variable substitution](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/features/secrets.adoc#kubernetes-secrets), but extremely tedious. Will definitely require adding some custom template logic to our wrapper chart to make this practical.

---

[Encrypted credentials](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/featuresthe user loses the ability to properly track credentials in git using a standard git encryption tool. Also "Encryption is done using the Jenkins-internal secret key which is unique for every Jenkins instance. It means that the credentials are not portable between instances" is really concerning.

---

The kubernetes credentials plugin looks pretty promising, but it has a major issue that prevents it being used in most migration scenarios. The Credential ID comes from the secret's name, which means that the credential IDs:

- must fit the kubernetes limitations for secret names
- must not clash with whatever arbitrary other secrets exist in the namespace

If the secret name had been supplied as either an annotation value or a data value, this wouldn't be a problem, as safe secret names could be chosen independently of the credential name. but as it stands, in order to adopt this plugin, a user would need to potentially rename credentials and rewrite any pipelines that consume them, which is not always practical.

One uninvestigated possibility to mitigate only the second of the listed issues is to put the credentials in a separate namespace, although that would then require giving jenkins cluster-wide secret access, which would be an unacceptable security risk in most cases.

Reply:
`controller.additionalSecrets` is mounted onto the Jenkins file system and you can then reference them in your jcasc file like:

`password: ${my-mounted-secret-name}`

I can't see why 'must fit the kubernetes limitations for secret names' matters, as you can use whatever secret name 

I missed this docs before but it should explain it better:
https://github.com/jenkinsci/helm-charts/tree/main/charts/jenkins#additional-secrets

I still don't link this is needed as `controller.additionalSecrets` seems to fully handle your usecase

Reply:
@timja should we close this then?

Reply:
Yes